core/message-tags-3.3.md

```diff
@@ -105,7 +105,8 @@ Servers MAY apply moderation to this command using existing or newly specified m
 
 Servers MUST NOT deliver `TAGMSG` to clients that haven't negotiated the message tags capability.
 
-Servers SHOULD reject any `TAGMSG` command sent without tags. In this case, they MUST use the `ERR_NEEDMOREPARAMS` (`461`) error numeric.
+Servers SHOULD reject any `TAGMSG` command sent without tags, or where all tags on the message end up
+being removed by the server. In this case, they MUST use the `ERR_NEEDMOREPARAMS` (`461`) error numeric.
```

Comment:
Yeah, MUST is fine. When this was originally written there wasn't anything in the spec that meant the servers had to inspect tag data for size limits etc so it was more lenient on servers that wanted to be lazy. That's not an option now.